src/main/java/org/terasology/climateConditions/ExtremeClimateConditionSystem.java

```diff
@@ -0,0 +1,198 @@
+        if (event.getActionId().equals(FROSTBITE_DAMAGE_ACTION_ID)) {
+            for (EntityRef entity : entityManager.getEntitiesWith(LocationComponent.class, AliveCharacterComponent.class)) {
+                    // Check to see if health should be decreased
+                    LocationComponent location = entity.getComponent(LocationComponent.class);
+                    float height = location.getLocalPosition().getY();
+                    //   float deltaHeight = location.getLastPosition().getY();
+                    // float lastHeight = height - deltaHeight;
+                    if (height > thresholdHeight) {
+                        Prefab frostbiteDamagePrefab = prefabManager.getPrefab("ClimateConditions:FrostbiteDamage");
+                        entity.send(new DoDamageEvent(healthDecreaseAmount, frostbiteDamagePrefab));
+
+                            EntityRef particleEntity = entityManager.create("climateConditions:snowParticleEffect");
+                            LocationComponent targetLoc = entity.getComponent(LocationComponent.class);
+                            LocationComponent childLoc = particleEntity.getComponent(LocationComponent.class);
+                            childLoc.setWorldPosition(targetLoc.getWorldPosition());
+                            Location.attachChild(entity, particleEntity);
+                            particleEntity.setOwner(entity);
+                            Vector3f direction = targetLoc.getLocalDirection();
+                            direction.normalize();
+
+                            //NOTE: initializing velocity using the constructor wasn't working hence the long method.
+                            //particleEntity.addComponent(new VelocityRangeGeneratorComponent(direction,direction.scale(2)));
+                            VelocityRangeGeneratorComponent velocity = particleEntity.getComponent(VelocityRangeGeneratorComponent.class);
+                            direction.scale((float)0.5);
+                            direction.addY((float)0.5);
+                            velocity.minVelocity = direction;
+                            direction.scale((float)1.5);
+                            velocity.maxVelocity = direction;
+                            particleEntity.addOrSaveComponent(velocity);
+                            if(!entity.hasComponent(SnowParticleComponent.class)) {
+                                entity.addComponent(new SnowParticleComponent());
+                            }
+                            entity.getComponent(SnowParticleComponent.class).particleEntity = particleEntity;
+                    }
+            }
+        }
+    }
+
+    @ReceiveEvent(priority = EventPriority.PRIORITY_NORMAL + 10 )
+    public void onHealthRegen(ActivateRegenEvent event, EntityRef player, LocationComponent location) {
+        float height = location.getLocalPosition().getY();
+        if(height > thresholdHeight){
+            player.send(new DeactivateRegenEvent());
```

Comment:
First question: is `ActivateRegenEvent` consumable? If so, it might be easier to just consume it before it reaches other event handlers, instead of deactivating the regeneration right away.

Second question/remark: I think this will deactivate all kinds of regeneration, including the effect of drinking potions or similar. This might be desired, but is probably something we should keep in mind as it might need some adjustment later on.

Reply:
I'll just ask what consumable means and if it is how to consume it on the channel 😅
And about the deactivate regen, I think yes it will deactivate all kinds of regen..  And I don't think it is desirable as potions need to be functional even in the extreme climates... I don't think nature should be enough to disallow the use of magical potions. Definitely need a filter on the kind of regen being received.

Reply:
Ookay so ActivateRegenEvent is not consumable.